packages/nocodb/src/plugins/gcs/Gcs.ts

```diff
@@ -176,13 +177,27 @@ export default class Gcs implements IStorageAdapterV2 {
 
     const [url] = await this.storageClient
       .bucket(this.bucketName)
-      .file(key)
+      .file(this.patchKey(key))
       .getSignedUrl(options);
 
     return url;
   }
 
-  public async scanFiles(_globPattern: string): Promise<Readable> {
-    return Promise.resolve(undefined);
+  public async scanFiles(globPattern: string): Promise<Readable> {
+    const stream = new Readable({
+      objectMode: true,
+      read() {},
+    });
+
+    const [files] = await this.storageClient.bucket(this.bucketName).getFiles({
+      prefix: globPattern.replace(/^\*/, ''),
+    });
+
+    for (const file of files) {
+      stream.push(file.name);
+    }
+    stream.push(null);
```

Comment:
To address @mertmit's comment:
```suggestion
    this.storageClient.bucket(this.bucketName).getFilesStream({
      prefix: globPattern.replace(/^\*/, ''),
      autoPaginate: true,
    })
      .on('error', (err) => {
        stream.destroy(err);
      })
      .on('data', (file) => {
        stream.push(file.name);
      })
      .on('end', () => {
        stream.push(null);
      });
```